src/main/java/com/vmware/vipclient/i18n/messages/api/opt/local/LocalMessagesOpt.java

```diff
@@ -62,11 +62,14 @@ public void getComponentMessages(MessageCacheItem cacheItem) {
 				path = Paths.get(uri);
 				messages = JSONBundleUtil.getMessages(path);
 			}
-			cacheItem.setCacheItem(messages, null, System.currentTimeMillis(), null);
```

Comment:
Why move this out of the try code block?

Reply:
Only update the timestmap if the fetch was successful. However, a try catch inside FileUtil.readJson may hide a failure scenario. Checking if messages is not empty would confirm that the fetch was successful.